packages/docusaurus-mdx-loader/src/loader.ts

```diff
@@ -173,7 +173,7 @@ export async function mdxLoader(
       ...(reqOptions.beforeDefaultRemarkPlugins ?? []),
       ...getAdmonitionsPlugins(reqOptions.admonitions ?? false),
       ...DEFAULT_OPTIONS.remarkPlugins,
-      ...(reqOptions.markdownConfig.mermaid ? [mermaid] : []),
+      ...(reqOptions.markdownConfig?.mermaid ? [mermaid] : []),
```

Comment:
This will be flagged by type-aware linting as unnecessary; we need to mark `markdownConfig` as optional in the options type. I'm not sure all this is necessary, though; `markdownConfig` is definitely required into the future.

Reply:
> We don't any anything being reported afaik, is this linting active?

No, but I occasionally turn it on and fix reports locally. It's quite slow so I'm hesitant to add it to the CI.

> I'm ok to just have this in v2, and remove this asap in v3 on the mdx 2 branch? We don't have to keep it, and it can be a temporary thing

That's reasonable, but people relying on internal APIs and coercing us to make changes sounds like a bad precedent. I'm okay to accept the change though.

Reply:
👍 

Let's say it's not a commitment to always do this, and only a best effort we can do in case a small change from us can prevent a few plugins to break

Reply:
OK, but the type of `Options.markdownConfig` still needs to be marked as optional ^^

Reply:
I'd rather not to. Somehow it's not optional, and we want users eventually using MDXLoaderOptions to be forced to provide it appropriately, and yet allow JS users to not have any runtime users. That sometimes makes sense to program defensively assuming all users (plugin authors here) won't use TS.

Reply:
That makes sense 👍